### PR TITLE
{tools}[GCCcore/12.2.0] datalad v0.18.4

### PR DESCRIPTION
**Pull request by jkuhl-uni**
Hi, this was the goal of the previous PRs:
- #18645
- #18672 
- #18673  
- #18674
With the bundle of these 5 PRs, the tool Datalad (https://www.datalad.org/) can be installed via EasyBuild.

### CODE REVIEW

**Comment by jkuhl-uni** [comment]
Hey @smoors,
could you please have a look at this again? Seems like the tests finally work :)

File: easybuild/easyconfigs/d/datalad/datalad-0.18.4-GCCcore-12.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+builddependencies = [
+    ('binutils', '2.39')
+]
+
+dependencies = [
+    ('git-annex', '10.20230802'),
+    ('Python', '3.10.8'),
+    ('tqdm', '4.64.1'),
+    ('PLY', '3.11'),
+    ('Rust', '1.65.0'),
```
Comment (smoors) [inline easybuild/easyconfigs/d/datalad/datalad-0.18.4-GCCcore-12.2.0.eb:23]: why is Rust needed as a dependency? if needed to build some of the python package, it should be a builddep.

**Comment by jkuhl-uni** [comment]
Done, anything else?

**Comment by smoors** [comment]
> Done, anything else?

can you also explain why Rust is needed?

**Comment by jkuhl-uni** [comment]
Hey there, I'm sorry it took that long, but it seems like it works without Rust as a dep or as a builddep.

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

File: easybuild/easyconfigs/d/datalad/datalad-0.18.4-GCCcore-12.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+use_pip = True
+
+exts_list = [
+    ('humanize', '3.13.1', {
+        'checksums': ['12f113f2e369dac7f35d3823f49262934f4a22a53a6d3d4c86b736f50db88c7b'],
+    }),
+    ('fasteners', '0.18', {
+        'checksums': ['cb7c13ef91e0c7e4fe4af38ecaf6b904ec3f5ce0dda06d34924b6b74b869d953'],
+    }),
+    ('patoolib', '1.12', {
```
Comment (smoors) [inline easybuild/easyconfigs/d/datalad/datalad-0.18.4-GCCcore-12.2.0.eb:35]:
```suggestion
    ('patool', '1.12', {
```

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by jkuhl-uni** [comment]
Hi, with all the changes, now python cannot import patool any more. I think that was why I used the source templates in the first place... I just changed the gitlab extension in the same way, but I think it has the same problem.
If you run python3 -c "import patoolib" and python3 -c "import gitlab" it should work :)

**Comment by smoors** [comment]
> Hi, with all the changes, now python cannot import patool any more. I think that was why I used the source templates in the first place... I just changed the gitlab extension in the same way, but I think it has the same problem. If you run python3 -c "import patoolib" and python3 -c "import gitlab" it should work :)

in that case you should add a custom modulename e.g.:
`'modulename': 'patoolib',`

**Comment by jkuhl-uni** [comment]
Thanks, I was not aware of that option

**Comment by jkuhl-uni** [comment]
Works for me, thank you for your help!

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen2

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @jkuhl-uni!

**Pull request merged**